### PR TITLE
kernel, load_self: Fix for empty tls

### PR DESCRIPTION
**Pull request by bookmist**

### CODE REVIEW

**Review by pent0 — APPROVED** [review]
Your solution is more thorough, thanks

File: vita3k/kernel/src/thread/thread.cpp [hunk 1]
```diff
@@ -107,18 +107,18 @@ SceUID create_thread(Ptr<const void> entry_point, KernelState &kernel, MemState
...
-    write_tpidruro(*thread->cpu, user_tls_address_ptr.address());
-
     if (kernel.tls_address) {
-        assert((kernel.tls_psize == 0xcccccccc ? 0 : kernel.tls_psize) <= (kernel.tls_msize == 0xcccccccc ? 0 : kernel.tls_msize));
+        auto user_tls_address_ptr = base_tls_address_ptr + kernel_tls_size;
+        write_tpidruro(*thread->cpu, user_tls_address_ptr.address());
+        assert(kernel.tls_psize <= kernel.tls_msize);
         memcpy(user_tls_address_ptr.get(mem), kernel.tls_address.get(mem), kernel.tls_psize);
+    } else {
+        write_tpidruro(*thread->cpu, 0);
```
Comment (sunho) [inline vita3k/kernel/src/thread/thread.cpp:121]: We should set kernel tls address here.

Comment (sunho) [reply vita3k/kernel/src/thread/thread.cpp:121]:
```
auto user_tls_address_ptr = base_tls_address_ptr + kernel_tls_size;
write_tpidruro(*thread->cpu, user_tls_address_ptr.address());
```
Just move this out and delete else. Then, looks fine!

Comment (bookmist) [reply vita3k/kernel/src/thread/thread.cpp:121]: No, we should not, We have zero size tls, so I set null to tpidruro to crash on unexpected tls using.

Comment (sunho) [reply vita3k/kernel/src/thread/thread.cpp:121]: If we lle some library that access kernel tls through tpidruro, we can know immediately in your way.

Comment (sunho) [reply vita3k/kernel/src/thread/thread.cpp:121]: LGTM

**Pull request merged**